wgengine/userspace.go

```diff
@@ -175,6 +176,13 @@ type Config struct {
 	// If nil, a fake OSConfigurator that does nothing is used.
 	DNS dns.OSConfigurator
 
+	// ReconfigureVPN provides a hook for platforms like Android to know when
+	// it's time to reconfigure their VPN implementation. Such
+	// platforms can only configure their VPN at startup of the VPN, so they
+	// prefer only to do so when a major operation like a link change has
+	// happened.
```

Comment:
// Such platforms can only set their entire VPN configuration (routes, DNS, etc)
// at all once and can't make piecemeal incremental changes so this provides
// a hook to "flush" a batch of Router and/or DNS changes.